src/js/components/Form/index.d.ts

```diff
@@ -1,16 +1,17 @@
 import * as React from "react";
 
-export interface FormProps {
+export interface FormProps<T> {
```

Comment:
Why do you think the <T> is essential for this solution?

Reply:
Being able to type your input makes it possible to type check your handler function -- consider:

```typescript
interface MyFormState {
  email: string;
}

function onChange(value: OtherFormState): void {
  const name = value.name; // no error here
}

// elsewhere
<Form onChange={onChange}>
  <FormField name="email" label="Email">
        <TextInput name="email" />
      </FormField>
</Form>
```
Without the generic, this code will not raise an error, however the `onChange` function is expecting the wrong shape.

Ideally we would also add type definitions such that if you tried to use the wrong `name` parameter in `FormField` or `TextInput`, this would also raise an error, but that is beyond the scope of this change, IMO.